src/python/pants/backend/python/subsystems/pex_bootstrap/_ptex_launcher.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+"""Entrypoint script for a .ptex file generated with --generate-ptex.
+
+This script will build a .ipex file into a temporary directory, then execute it. If
+PANTS_PTEX_HYDRATE_ONLY_TO_IPEX=<filename> is set, however, the script will generate the ipex at
+<filename>, then exit without executing the .ipex file.
+"""
+
+import json
+import os
+import sys
+import tempfile
+
+from pex import resolver
+from pex.common import open_zip
+from pex.fetcher import Fetcher, PyPIFetcher
+from pex.interpreter import PythonInterpreter
+from pex.interpreter_constraints import matched_interpreters
+from pex.pex_builder import PEXBuilder
+from pex.pex_info import PexInfo
+
+
+APP_CODE_PREFIX = 'user_files/'
+
+
+def _strip_app_code_prefix(path):
+  if not path.startswith(APP_CODE_PREFIX):
+    raise ValueError(f"Path {path} in PTEX-INFO did not begin with '{APP_CODE_PREFIX}'.")
+  return path[len(APP_CODE_PREFIX):]
+
+
+HYDRATE_ONLY_NO_EXEC_ENV_VAR = 'PANTS_PTEX_HYDRATE_ONLY_TO_IPEX'
+
+
+def main(self):
+  filename_base, _ext = os.path.splitext(self)
+
+  # Build a .ipex file in a new temporary dir, each time.
+  td = tempfile.mkdtemp()
+  hydrated_ipex_file = os.environ.get(HYDRATE_ONLY_NO_EXEC_ENV_VAR, None)
+  oneshot_run = False
```

Comment:
Rather than `oneshot_run` I'd favour a name like `do_run` (because the oneshot-ness isn't actually relevant here - whether we run after generating to a stable path or not isn't relevant, and it leaves me asking what the oneshot means).

Reply:
Yes! I've changed it to `whether_to_run` -- not sure of your feelings on that vs `do_run`?